### PR TITLE
Side drawer for user shopping list.

### PR DESCRIPTION
**Pull request by inquisitiv-e**
- Added Drawer for Shopping List
- Added ShoppingList backend
- Added SnackBar, that appears upon adding an item to shopping list. 

Screen Shot 2019-07-23 at 10 43 18 PM

- [x] Save the checklist item in db (Now done!)

### CODE REVIEW

File: imports/ui/components/SnackBar.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setState({ open: false });
+  };
+  render() {
+    return (
+      <div>
+        <IconButton
+          style={{ display: this.state.isFav ? "none" : "" }}
```
Comment (nicolehli) [inline imports/ui/components/SnackBar.jsx]: it's nice to have code to reuse. But should rename things like isFav to make sure code makes sense in new places.

File: imports/ui/components/ShoppingListButton.jsx [hunk 2]
```diff
@@ -1,18 +1,159 @@
...
+                    {" " + text.location.address}
+                  </React.Fragment>
+                }
+              />
+              <ListItemSecondaryAction>
+                <Checkbox
+                  edge="end"
+                  onChange={this.handleToggle(index)}
+                  checked={this.state.checked.indexOf(index) !== -1}
+                  inputProps={{ "aria-labelledby": labelId }}
```
Comment (nicolehli) [inline imports/ui/components/ShoppingListButton.jsx]: Do we want the checkboxed checked to persist after refresh?

File: imports/ui/components/Card.jsx [hunk 3]
```diff
@@ -110,22 +103,8 @@ class CardComponent extends Component {
...
-              </IconButton>
-              <IconButton
-                style={{ display: this.state.isFav ? "" : "none" }}
-                onClick={this.onFavPressed}
-              >
-                <CheckedIcon />
-              </IconButton>
-            </div>
-          )}
+          {/* Favorite icon is a part of the stretch goal to add wishlist, use later */}
```
Comment (nicolehli) [inline imports/ui/components/Card.jsx]: I think this comment should be removed in the refactor PR that I did. Maybe during merge conflict didn't pick the incoming change?

File: imports/ui/components/Card.jsx [hunk 4]
```diff
@@ -110,22 +103,8 @@ class CardComponent extends Component {
...
+          {this.props.currentUser && <AddShoppingList item={this.state.data} />}
```
Comment (nicolehli) [inline imports/ui/components/Card.jsx]: Add to shopping list could add item multiple times when google between + icon and ✔️icon.  Also going from + to ✔️  doesn't remove the item from the shopping cart list. Do we want a way to remove items from shopping cart?

File: imports/ui/components/ShoppingListButton.jsx [hunk 5]
```diff
@@ -1,18 +1,159 @@
...
+import "./ShoppingListButton.css";
 
 class ShoppingListButton extends Component {
   constructor(props) {
     super(props);
+    this.state = { left: false, checked: [] };
+  }
+
+  componentDidMount() {
+    this.props.getShoppingListItems();
```
Comment (nicolehli) [inline imports/ui/components/ShoppingListButton.jsx]: When I log out of user1 and log in as user2, I see user1's shopping list.

File: imports/ui/components/ShoppingListButton.jsx (hunk 2 above)
Comment (nicolehli) [reply imports/ui/components/ShoppingListButton.jsx]: Do we want snackbar to stay open when we check boxes? Right now it collapses every time a box is checked.

File: imports/ui/components/ShoppingListButton.jsx (hunk 5 above)
Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: Unfortunately I cant do anything about that right now :/ I dont have access to on log out

Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: You need to refresh the website before viewing your own

File: imports/ui/components/SnackBar.jsx (hunk 1 above)
Comment (inquisitiv-e) [reply imports/ui/components/SnackBar.jsx]: Renamed

File: imports/ui/components/ShoppingListButton.jsx (hunk 2 above)
Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: Checkbox now persist in database.

Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: > Do we want snackbar to stay open when we check boxes? Right now it collapses every time a box is checked.

I think you meant shoppinglist drawer right? I fixed that in the new commit where the drawer doesnt collapse when you check items.

File: imports/ui/components/Card.jsx (hunk 4 above)
Comment (inquisitiv-e) [reply imports/ui/components/Card.jsx]: #76

File: imports/ui/components/ShoppingListButton.jsx (hunk 5 above)
Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: Resolved now

File: imports/ui/components/Card.jsx (hunk 3 above)
Comment (inquisitiv-e) [reply imports/ui/components/Card.jsx]: Removed.

File: imports/ui/components/ShoppingListButton.jsx [hunk 6]
```diff
@@ -1,18 +1,149 @@
...
+
   render() {
+    const items = this.props.items.shoppingList;
     return (
-      <span>{this.props.currentUser && 
-      <Button>ShoppingList</Button>}</span>
+      <span>
+        {this.props.currentUser && (
+          <span>
+                    <Button
```
Comment (nicolehli) [inline imports/ui/components/ShoppingListButton.jsx]: Did you run prettier?

Comment (inquisitiv-e) [reply imports/ui/components/ShoppingListButton.jsx]: Yeah but it didn't fix. I'll move it back a little bit and then run prettier again.

**Pull request merged**